requests_auth/authentication.py

```diff
@@ -142,6 +142,7 @@ def __init__(self, token_url: str, username: str, password: str, **kwargs):
         :param password: Resource owner password.
         :param timeout: Maximum amount of seconds to wait for a token to be received once requested.
         Wait for 1 minute by default.
+        :param early_expire: Amount of seconds before the actual expiry to renew the token early. 5 seconds by default.
```

Comment:
I actually have another project named `httpx_auth` (same but for httpx library instead of requests). And this is already implemented. I would want to keep the same names, documentation and default values. Could you remove this feature from this PR to keep things separate? You are welcome to come with an additional PR for the early expiry of course :)

Thanks